kubernetes/resource_kubernetes_resource_priority_class.go

```diff
@@ -0,0 +1,227 @@
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	api "k8s.io/api/scheduling/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgApi "k8s.io/apimachinery/pkg/types"
+	kubernetes "k8s.io/client-go/kubernetes"
+)
+
+func resourceKubernetesPriorityClass() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesPriorityClassCreate,
+		Read:   resourceKubernetesPriorityClassRead,
+		Exists: resourceKubernetesPriorityClassExists,
+		Update: resourceKubernetesPriorityClassUpdate,
+		Delete: resourceKubernetesPriorityClassDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metadata": metadataSchema("priority class", true),
+			"description": {
+				Type:        schema.TypeString,
+				Description: "An arbitrary string that usually provides guidelines on when this priority class should be used.",
+				Optional:    true,
+				Default:     "",
+			},
+			"global_default": {
+				Type:        schema.TypeBool,
+				Description: "Specifies whether this PriorityClass should be considered as the default priority for pods that do not have any priority class. Only one PriorityClass can be marked as `globalDefault`. However, if more than one PriorityClasses exists with their `globalDefault` field set to true, the smallest value of such global default PriorityClasses will be used as the default priority.",
+				Optional:    true,
+				Default:     false,
+			},
+			"value": {
+				Type:        schema.TypeInt,
+				Description: "The value of this priority class. This is the actual priority that pods receive when they have the name of this class in their pod spec.",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceKubernetesPriorityClassCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	value := d.Get("value").(int)
+	description := d.Get("description").(string)
+	globalDefault := d.Get("global_default").(bool)
+
+	priorityClass := api.PriorityClass{
+		ObjectMeta:    metadata,
+		Description:   description,
+		GlobalDefault: globalDefault,
+		Value:         int32(value),
+	}
+
+	log.Printf("[INFO] Creating new priority class: %#v", priorityClass)
+	out, err := conn.Scheduling().PriorityClasses().Create(&priorityClass)
+	if err != nil {
+		return fmt.Errorf("Failed to create priority class: %s", err)
+	}
+	log.Printf("[INFO] Submitted new priority class: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
+		createdPriorityClass, err := conn.Scheduling().PriorityClasses().Get(out.Name, meta_v1.GetOptions{})
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		if createdPriorityClass.Value == priorityClass.Value {
```

Comment:
Is that really needed? It seems this was ported from _Resource Quota_ which was quite a special case that was implemented more than two years ago.

Reply:
Removed this.